browser/brave_webcompat_reporter/webcompat_reporter_service.cc

```diff
@@ -0,0 +1,102 @@
+
+void BraveWebcompatReporterService::SubmitReport(std::string report_domain) {
+  std::string api_key = brave::GetAPIKey();
+
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  std::string upload_url;
+  env->GetVar("BRAVE_WEBCOMPAT_REPORT_ENDPOINT", &upload_url);
+  if (upload_url.empty())
+    upload_url = kBraveWebcompatReportEndpoint;
+
+  base::Value post_data_obj(base::Value::Type::DICTIONARY);
+  post_data_obj.SetKey("domain", base::Value(report_domain));
+  post_data_obj.SetKey("api_key", base::Value(api_key));
+
+  std::string post_data;
+  base::JSONWriter::Write(post_data_obj, &post_data);
+
+  base::PostTaskWithTraits(FROM_HERE, {content::BrowserThread::UI},
+      base::BindOnce(&BraveWebcompatReporterService::CreateAndStartURLLoader,
+          base::Unretained(this), upload_url, post_data));
+}
+
+void BraveWebcompatReporterService::CreateAndStartURLLoader(
+    const std::string& upload_url,
+    const std::string& post_data) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+
+  std::string content_type = "application/json";
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  resource_request->url = GURL(upload_url);
+  resource_request->method = "POST";
+  resource_request->load_flags = net::LOAD_DO_NOT_SEND_COOKIES |
+                                 net::LOAD_BYPASS_CACHE |
+                                 net::LOAD_DISABLE_CACHE |
+                                 net::LOAD_DO_NOT_SEND_AUTH_DATA;
+
+  net::NetworkTrafficAnnotationTag traffic_annotation =
+      net::DefineNetworkTrafficAnnotation("background_performance_tracer", R"(
+        semantics {
+          sender: "Brave Web Compatibility Reporting"
+          description:
+            "A user-initiated report of a website that appears broken as a"
+            "result of having Brave Shields enabled."
+          trigger:
+            "Though the 'Report a Broken Site' option of the help menu or"
+            "the Brave Shields panel."
+          data: "Broken domain and IP address."
+          destination: OTHER
+          destination_other: "Brave developers"
+        }
+        policy {
+          cookies_allowed: NO
+        })");
+
+  simple_url_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), traffic_annotation);
+  simple_url_loader_->AttachStringForUpload(post_data, content_type);
+  simple_url_loader_->DownloadToStringOfUnboundedSizeUntilCrashAndDie(
+      g_browser_process->shared_url_loader_factory().get(),
+      base::BindOnce(&BraveWebcompatReporterService::OnSimpleURLLoaderComplete,
+                     base::Unretained(this)));
+}
+
+void BraveWebcompatReporterService::OnSimpleURLLoaderComplete(
+    std::unique_ptr<std::string> response_body) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+
+  bool success = !!response_body;
+
+  if (success) {
+    LOG(INFO) << "Successfully uploaded webcompat report. Thanks!" << std::endl;
```

Comment:
This isn't providing any useful information that the user will see. Shouldn't it be reporting success/failure back to the dialog?

Reply:
@alexwykoff said this should be fine for the MVP

Reply:
The spec does refer to a 'Thank you' dialog, but it does not concern itself with network state. If the network is down, it is presumed that there are more obvious conditions the user would experience. (Ex: 'No Internet' dino screen)

Reply:
@alexwykoff there are plenty of other possible ways that it could fail including issues with our endpoint

Reply:
Totally right, but to the mass market end consumer, they are total gibberish. 
Developer console log makes sense, but not a custom UI.